internal/fs/inode/file.go

```diff
@@ -471,6 +476,29 @@ func (f *FileInode) Write(
 	return
 }
 
+func (f *FileInode) ensureBuffer(bufferSize int) {
+	if f.buffer != nil {
+		return
+	}
+	if bufferSize <= 50 {
+		f.buffer = &buffer.InMemoryWriteBuffer{}
```

Comment:
Will this throw away the data accumulated in the previous write-buffer?

Reply:
this is stored at inode level. so until the inode is retained, the data will not be thrown.